src/create_disk.sh

```diff
@@ -574,14 +594,14 @@ if [[ ${secure_execution} -eq 1 ]]; then
     # set up dm-verity for the rootfs and bootfs
     create_dmverity root $rootfs
     create_dmverity boot $rootfs/boot
-
     # We need to run the genprotimg step in a separate step for rhcos release images
     if [ ! -e /dev/disk/by-id/virtio-genprotimg ]; then
         echo "Building local Secure Execution Image, running zipl and genprotimg"
         # run zipl with root hashes as kargs
         rdcore_zipl_args+=("--secex-mode=enforce" "--hostkey=/dev/disk/by-id/virtio-hostkey")
         rdcore_zipl_args+=("--append-karg=rootfs.roothash=$(cat /tmp/root-roothash)")
         rdcore_zipl_args+=("--append-karg=bootfs.roothash=$(cat /tmp/boot-roothash)")
+        rdcore_zipl_args+=("--append-file=/tmp/ignition.asc")
```

Comment:
So continuing the discussion from https://github.com/coreos/fedora-coreos-config/pull/1939#discussion_r1104845076 here, having the private key live in `/tmp` doesn't feel right. Everything runs as root in the initrd anyway, but `/tmp` is shared space that other applications read from and write to routinely. E.g. you can imagine some sloppy script doing a `cat` and it ending up on the console.

For the external genprotimg path, it's easier since we do the appending manually.

For the `rdcore zipl` path, I think doing this properly would require teaching `--append-file` to have different source and dest paths (e.g. `--append-file /tmp/ignition.asc:/usr/lib/coreos/ignition.asc`). But short-term, I think we can hack this by adding a bind mount from `/tmp/ignition.asc` to `${deploy_root}/usr/lib/coreos/ignition.asc` and then here we can pass `--append-file=/usr/lib/coreos/ignition.asc`.

If that doesn't work, I'm OK to defer on this for now.

Reply:
I guess for now we can leave an empty file, indeed this build is only possible locally for dev-purposes. Official builds would have private key attached by a script, so realroot is not touched at all. I like an idea to teach `rdcore`  how to `--append-file path1:path2`, so will add it my backlog.